### ISSUE TITLE
Request: Add option to ignore deprecation warnings in GoReleaser check

### ISSUE THREAD

**Issue by ryancurrah** [goreleaser/goreleaser#4179]
*Not Urgent - More of a nice to have, I can work around it by getting an older release off GitHub releases*

### Is your feature request related to a problem? Please describe.

I'm currently working on an upgrade process that involves the automated modification of more than 300 GoReleaser configuration files using yq, with subsequent validation via goreleaser check. However, I've encountered an issue that disrupts this process.

When running the check, it errors out due to the deprecated `brews.tap` usage. This is a deprecated configuration that I am aware of and am planning to resolve by updating to GoReleaser 1.19 (in CI). However, I cannot proceed with the update until I've addressed the deprecation of `archive.replacements`.

I can't use the latest GoReleaser version locally to push out this change because the `brews.tap` deprecation error occurs when running `goreleaser check` to validate that my changes to remove `archive.replacements` with a change to `name_template` are valid.

I'm looking forward to hearing your thoughts on this proposal. Thank you for considering this enhancement.

```
❯ goreleaser check
  • by using this software you agree with its EULA, available at https://goreleaser.com/eula
  • running goreleaser v1.19.2-pro
  • loading config file                              file=.goreleaser.yml
  • importing .goreleaser.common.yml
  • checking config...
    • DEPRECATED: `brews.tap` should not be used anymore, check https://goreleaser.com/deprecations#brewstap for more info
  •                                                  error=configuration is valid, but uses deprecated properties
  ⨯ command failed                                   error=1 out of 1 configuration file(s) have issues
```

### Describe the solution you'd like

My suggestion is to introduce an option in GoReleaser check that would allow the ignoring or skipping of deprecation warnings. This would provide greater flexibility in managing updates and deprecations, particularly for those with a significant number of GoReleaser configuration files.

### Describe alternatives you've considered

Manually install an older version of goreleaser from GitHub on my laptop instead of using Homebrew. As homebrew does not let me choose a version to install.

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [X] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
thanks, working on this to release as a patch on v1.19.

**Comment by caarlos0** [comment]
PS: you can already check for the exit code, deprecation errors exit 2, other errors exit 1.

**Comment by ryancurrah** [comment]
Oh did not know that. Its good enough for me to check exit code. We can close this if you want.

**Comment by ryancurrah** [comment]
For others doing something similar here is a script we are running on all our go release repos. Turns out we now have 568 repos using Go Releaser 😎 .

```
#!/usr/bin/env bash
set -e

cd "${1}"

# When brews exists
if [ $(yq e '.brews != null' .goreleaser.yml) = "true" ]; then
    # Tap is depreciated in favor of repository
    yq -i '.brews[].repository = .brews[].tap | del(.brews[].tap)' .goreleaser.yml
fi

# When archives exists
if [ $(yq e '.brews != null' .goreleaser.yml) = "true" ]; then
    # Replacements is depreciated in favor of using name_template
    yq -i 'del(.archives[].replacements)' .goreleaser.yml

    # Update name_template
    yq -i '.archives[].name_template = "{{ replace .Binary \"_\" \"-\" }}_{{ .Version }}_{{ title .Os }}_{{- if eq .Arch \"amd64\" }}x86_64{{- else }}{{ .Arch }}{{ end }}{{- if .Arm }}v{{ .Arm }}{{ end }}"' .goreleaser.yml

    # Archives format_overrides for Windows to use zip instead
    yq -i '.archives[].format_overrides = [{"goos": "windows", "format": "zip"}]' .goreleaser.yml
fi

# Ensure config is valid
set +e
goreleaser check
if [ "${?}" = 1 ];
then
    exit 1
fi
```

**Comment by caarlos0** [comment]
added a `--soft` flag to pro to make it easier, will be released in the next version!

thanks!
